virtool/labels/api.py

```diff
@@ -1,180 +1,162 @@
+from typing import List, Union
+
 from aiohttp.web_exceptions import HTTPBadRequest, HTTPNoContent
+from aiohttp_pydantic import PydanticView
+from aiohttp_pydantic.oas.typing import r200, r201, r204, r400, r404
 from sqlalchemy import select
 from sqlalchemy.exc import IntegrityError
 from sqlalchemy.ext.asyncio import AsyncSession
 
 import virtool.http.routes
-import virtool.validators
 from virtool.api.response import EmptyRequest, NotFound, json_response
 from virtool.mongo.transforms import apply_transforms
-from virtool.http.schema import schema
 from virtool.labels.db import SampleCountTransform
-from virtool.labels.models import Label
+from virtool.labels.models import Label as LabelSQL
 from virtool.pg.utils import get_generic
-
+from virtool.labels.oas import CreateLabelSchema, EditLabelSchema
+from virtool_core.models.label import Label, LabelMinimal
 routes = virtool.http.routes.Routes()
 
 
-@routes.get("/labels")
-async def find(req):
-    """
-    Get a list of all label documents in the database.
+@routes.view("/labels")
+class LabelsView(PydanticView):
 
-    """
-    term = req.query.get("find")
+    async def get(self) -> Union[r200[List[LabelMinimal]], r400]:
+        """
+        List all sample labels.
 
-    statement = select(Label).order_by(Label.name)
-    if term:
-        statement = statement.filter(Label.name.ilike(f"%{term}%"))
+        Status Codes:
+            200: Successful operation
+            400: Invalid query
+        """
+        term = self.request.query.get("find")
 
-    labels = await get_generic(req.app["pg"], statement)
+        statement = select(LabelSQL).order_by(LabelSQL.name)
+        if term:
+            statement = statement.filter(LabelSQL.name.ilike(f"%{term}%"))
 
-    documents = await apply_transforms(
-        [label.to_dict() for label in labels], [SampleCountTransform(req.app["db"])]
-    )
+        labels = await get_generic(self.request.app["pg"], statement)
 
-    return json_response(documents)
+        documents = await apply_transforms(
+            [label.to_dict() for label in labels], [SampleCountTransform(self.request.app["db"])]
+        )
 
+        return json_response(documents)
 
-@routes.get("/labels/{label_id}")
-async def get(req):
-    """
-    Get a complete label document.
+    async def post(self, data: CreateLabelSchema) -> Union[r201[Label], r400]:
+        """
+        Create a sample label.
 
-    """
-    async with AsyncSession(req.app["pg"]) as session:
-        result = await session.execute(
-            select(Label).filter_by(id=int(req.match_info["label_id"]))
-        )
+        Colors must be valid hexadecimal code.
```

Comment:
Grammar. The color must be a valid hexadecimal code.

These docstrings are especially important because they will hopefully wind up in the web-based docs.